### PR TITLE
Adding NeMo 2.0 T5 finetuning (on Squad dataset)

### PR DESCRIPTION
**Pull request by huvunvidia**
# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

**Comment by JRD971000** [comment]
LGTM

**Review by JRD971000 — APPROVED** [review]

File: .github/workflows/cicd-main.yml [hunk 1]
```diff
@@ -5216,6 +5216,21 @@ jobs:
...
+    uses: ./.github/workflows/_test_template.yml
+    if: contains(fromJSON(needs.cicd-test-container-setup.outputs.test_to_run), 'L2_NeMo_2_T5_Finetuning') || needs.cicd-test-container-setup.outputs.all == 'true'
+    with:
+      RUNNER: self-hosted-azure
+      SCRIPT: |
+        NVTE_FUSED_ATTN=0 NVTE_FLASH_ATTN=0 python tests/collections/llm/megatron_t5_finetuning.py \
+        --devices=2 \
+        --max-steps=250 \
+        --experiment-dir=tests/collections/llm/t5_finetune_results/${{ github.run_id }} \
+        --checkpoint-path=/home/TestData/nlp/megatron_t5/220m/nemo2.0_t5_220m_150steps
```
Comment (ko3n1g) [inline .github/workflows/cicd-main.yml:5230]: do you read from this directory or do you write into it?

**Review by ko3n1g — COMMENTED** [review]
QQ

File: .github/workflows/cicd-main.yml (hunk 1 above)
Comment (huvunvidia) [reply .github/workflows/cicd-main.yml:5230]: Hi Oliver,
I read from "/home/TestData/nlp/megatron_t5/220m/nemo2.0_t5_220m_150steps".
And write results to "tests/collections/llm/t5_finetune_results/${{ github.run_id }}"

**Review by ko3n1g — APPROVED** [review]

**Pull request merged**